plugins/cache_promote/lru_policy.h

```diff
@@ -53,11 +55,12 @@ class LRUHash
   void
   init(char *data, int len)
   {
-    SHA_CTX sha;
+    EVP_MD_CTX *ctx = EVP_MD_CTX_new();
```

Comment:
Can't you just call EVP_Digest()?  There is talk about the function here (https://github.com/openssl/openssl/issues/7219) and the issues around heap vs stack allocation.

Reply:
EVP_Digest is basically just a wrapper and it uses heap too. If you worry about the heap allocation, we should keep the context object somewhere to reuse and that would require us to use these separated functions, I'm not sure if we can have a reusable context for this plugin though.

For now I just added a line to set EVP_MD_CTX_FLAG_ONESHOT to minimize the difference from EVP_Digest, but I'm also ok with replacing the lines with EVP_Digest if you strongly think that is better.